jpsonic-main/src/main/java/com/tesshu/jpsonic/dao/MediaFileDao.java

```diff
@@ -865,56 +870,96 @@ select count(id)
+    public void updateArtistSort(ArtistSortCandidate cand) {
+        if (cand.getTargetField() == TargetField.ARTIST && cand.getTargetType() == MediaType.DIRECTORY) {
+            template.update("""
+                    update media_file
+                    set artist_reading = ?, artist_sort = ?, music_index = ?
+                    where id = ?
+                    """, cand.getReading(), cand.getSort(), cand.getMusicIndex(), cand.getTargetId());
+        } else if (cand.getTargetField() == TargetField.ARTIST) {
             template.update("""
                     update media_file
                     set artist_reading = ?, artist_sort = ?
                     where id = ?
-                    """, candidate.getReading(), candidate.getSort(), candidate.getId());
-        } else if (candidate.getField() == CandidateField.ALBUM_ARTIST) {
+                    """, cand.getReading(), cand.getSort(), cand.getTargetId());
+        } else if (cand.getTargetField() == TargetField.ALBUM_ARTIST) {
             template.update("""
                     update media_file
                     set album_artist_reading = ?, album_artist_sort = ?
                     where id = ?
-                    """, candidate.getReading(), candidate.getSort(), candidate.getId());
-        } else if (candidate.getField() == CandidateField.COMPOSER) {
+                    """, cand.getReading(), cand.getSort(), cand.getTargetId());
+        } else if (cand.getTargetField() == TargetField.COMPOSER) {
             template.update("""
                     update media_file
                     set composer_sort = ?
                     where id = ?
-                    """, candidate.getSort(), candidate.getId());
+                    """, cand.getSort(), cand.getTargetId());
         }
     }
 
+    public void updateArtistSort(List<ArtistSortCandidate> cands) {
+        String updates = "update media_file ";
+        StringBuilder cols = new StringBuilder("set ");
+        String cond = "where id = :id";
+
+        List<Object> args = new ArrayList<>();
+        cands.forEach(cand -> {
+            switch (cand.getTargetField()) {
+            case ARTIST -> {
+                cols.append("artist_reading=?, artist_sort=?,");
+                args.add(cand.getReading());
+                args.add(cand.getSort());
+            }
+            case ALBUM_ARTIST -> {
+                cols.append("album_artist_reading=?, album_artist_sort=?,");
+                args.add(cand.getReading());
+                args.add(cand.getSort());
+            }
+            case COMPOSER -> {
+                cols.append("composer_sort=?,");
+                args.add(cand.getSort());
+            }
+            case UNKNOWN -> {
+            }
+            default -> {
+            }
+            }
+        });
+        args.add(cands.get(0).getTargetId());
+
+        String query = updates + cols.toString().replaceFirst(",$", "") + cond;
```

Comment:
Make whitespace replacement a little stricter(Maybe it doesn't work with Postgres)